src/main/java/types/HostCpuPinning.java

```diff
@@ -0,0 +1,58 @@
+package types;
+
+import org.ovirt.api.metamodel.annotations.Link;
+import org.ovirt.api.metamodel.annotations.Type;
+
+/**
+ * Type representing a physical CPU of a host with
+ * the current pinning status.
+ *
+ * @author Lucia Jelinkova <ljelinko@redhat.com>
+ * @since 4.5
+ * @date 15 Feb 2022
+ * @status added
+ */
+@Type
+public interface HostCpuPinning extends Identified {
+
+    /**
+     * The id of the socket the CPU belongs to.
+     *
+     * @author Lucia Jelinkova <ljelinko@redhat.com>
+     * @since 4.5
+     * @date 15 Feb 2022
+     * @status added
+     */
+    String socketId();
+
+    /**
+     * The id of the core the CPU belongs to.
+     *
+     * @author Lucia Jelinkova <ljelinko@redhat.com>
+     * @since 4.5
+     * @date 15 Feb 2022
+     * @status added
+     */
+    String coreId();
+
+    /**
+     * The id of the CPU.
+     *
+     * @author Lucia Jelinkova <ljelinko@redhat.com>
+     * @since 4.5
+     * @date 15 Feb 2022
+     * @status added
+     */
+    String cpuId();
+
+    /**
+     * A VM that has its virtual CPU pinned to this physical CPU.
+     *
+     * @author Lucia Jelinkova <ljelinko@redhat.com>
+     * @since 4.5
+     * @date 15 Feb 2022
+     * @status added
+     */
+    @Link Vm vm();
```

Comment:
I am thinking that a single vm might not be enough. If we have manually pinned more VMs to a single CPU, we probably need a list of VMs here.